### PR TITLE
Add references to plot_khat #1841

### PR DESCRIPTION
**Pull request by Weijun-H**
## Description
This PR adds references to the plot_khat function, addressing #1841 

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format

### CODE REVIEW

**Comment by Weijun-H** [comment]
Add references to plot_khat #1841

File: arviz/plots/khatplot.py [hunk 1]
```diff
@@ -81,21 +81,28 @@ def plot_khat(
     ax: axes, optional
         Matplotlib axes or bokeh figures.
     hlines_kwargs: dictionary, optional
-        Additional keywords passed to ax.hlines.
+        Additional keywords passed to
+        :func:`matplotlib.axes.Axes.hlines`.
```
Comment (roshnaeem) [inline arviz/plots/khatplot.py:85]:
```suggestion
       :meth:`matplotlib.axes.Axes.hlines`.
```

File: arviz/plots/khatplot.py [hunk 2]
```diff
@@ -81,21 +81,28 @@ def plot_khat(
...
-        check the plotting method of the backend.
+        These are kwargs specific to the backend being used, passed to
+        :func:`matplotlib.pyplot.subplots` or
+        :func:`bokeh.plotting.figure`.
     show : bool, optional
         Call backend show function.
     kwargs :
-        Additional keywords passed to ax.scatter.
+        Additional keywords passed to
+        :func:`matplotlib.axes.Axes.scatter`.
```
Comment (roshnaeem) [inline arviz/plots/khatplot.py:96]:
```suggestion
        :meth:`matplotlib.axes.Axes.scatter`.
```

**Comment by Weijun-H** [comment]
Update references

**Review by roshnaeem — APPROVED** [review]

**Comment by ColCarroll** [comment]
thanks, @Weijun-H !

**Pull request merged**